Correcting gStands-Alpha gross errors and adding functions for a more pleasant game on the Alpha version

ZeroTwo-gStands opened the pull request:
In my fixes, I opened each file and tried to get rid of the bugs in them. Added new materials, sounds, new functions. My fixes are not as good as the original code and most likely require editing. But I think this will work for my first fixes. Especially for the Alpha version.

ZeroTwo-gStands committed:
Update The World

291 lines: Changed the health issue system.
Line 302: A system thanks to which Dio will laugh when killing Stardust.
Line 312: a system due to which, when received by Dio, it plays some kind of sound.
412 line: Changed sound when Stand appears
Line 465: Dio is a vampire, faster and stronger than normal Stand owners.
604 line: Changed stop time.
666 line:
Changed part of the start time.
765 line: Changed "Control" time stop. (Pressing, conditions, and the like. I do not know how to describe it)
self.Barrage and self.Donut damage settings added.
1043 stitching: adding limited knives.
1111 line: Adding sounds when you turn off the Stand

ZeroTwo-gStands committed:
Update shared.lua

ZeroTwo-gStands committed:
Update shared.lua

ZeroTwo-gStands committed:
localization

In the translation, I added only the new necessary lines for my innovations.

ZeroTwo-gStands committed:
New sounds

I have added sounds.
Various sounds for Taunt and just sounds that I liked. (All of this is mentioned in the code)

ZeroTwo-gStands committed:
Non-faint materials.

I added here all the materials that were not there.
For Wheel of Fortune shots, for Emperor bullets

ZeroTwo-gStands committed:
Cable

Cable materials

ZeroTwo-gStands committed:
autorun

New recordings of sounds in resources and adding the necessary lines to gstands.lua

ZeroTwo-gStands committed:
entites

Lots of bug fixes and added damage settings

ZeroTwo-gStands committed:
entites

Lots of bug fixes and added damage settings

ZeroTwo-gStands committed:
Add files via upload

copperwarrior commented on lua/autorun/gstands.lua:
```diff
@@ -1,1802 +1,2066 @@
+CreateConVar( "gstands_stand_hurt_stands", 1, {FCVAR_REPLICATED, FCVAR_ARCHIVE}, "#gstands.general.standuserharm" ) 
+CreateConVar( "gstands_the_world_timestop_length", 10, {FCVAR_REPLICATED, FCVAR_ARCHIVE},"#gstands.general.wolength")
+CreateConVar( "gstands_star_platinum_timestop_length", 5, {FCVAR_REPLICATED, FCVAR_ARCHIVE},"#gstands.general.splength")
+CreateConVar( "gstands_the_world_next_timestop", 30, {FCVAR_REPLICATED, FCVAR_ARCHIVE},"#gstands.general.splength")
```
Proper translation keys need to be added to these.

copperwarrior commented on lua/autorun/gstands.lua:
```diff
@@ -1,1802 +1,2066 @@
+CreateClientConVar( "gstands_replacement_gstands_star_platinum", "models/player/starpjf/spjf.mdl", true, true, "" )
+
+CreateClientConVar( "gstands_barrage_legs", 6, true, true, "Amount of sets of arms for legs" ) 
+CreateClientConVar( "gstands_block_arms", 4, true, true, "Amount of sets of arms for legs" ) 
```
Where is this used?

ZeroTwo-gStands replied:
I made a separate setting for the amount for the Silver Chariot block arms and kBarrage legs.

copperwarrior commented on lua/autorun/gstands.lua:
```diff
@@ -1,1802 +1,2066 @@
+CreateConVar( "gstands_cream_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_dark_blue_moon_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_death_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_ebony_devil_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_emperor_heal", 800, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_empress_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_geb_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_hanged_man_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_hermit_purple_heal", 1000, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_hermit_purple_two_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_hierophant_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_high_priestess_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_horus_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_judgement_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_justice_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_khnum_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_kiss_from_a_rose_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_lovers_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_magicians_red_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_osiris_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_sethan_heal", 800, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_cream_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_strength_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_tenore_sax_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_the_fool_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_the_sun_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_thoth_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_tower_of_gray_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_wheel_of_fortune_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_yellow_temperence_heal", 1500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+
+CreateConVar( "gstands_star_platinum_leap", 1, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_the_world_leap", 1, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_cream_leap", 0.1, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+
+
+CreateConVar( "gstands_dark_blue_moon_range", 800, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_cream_range", 250, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_the_fool_range", 250, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
+CreateConVar( "gstands_star_platinum_range", 500, {FCVAR_REPLICATED, FCVAR_NONE}, "Is time stopped?" ) 
```
This will need a *much* cleaner solution. Something that iterates through each stand, as shown in the skin convars.

copperwarrior commented on lua/autorun/gstands.lua:
```diff
@@ -1,1802 +1,2066 @@
+	["models/gstands/stand_acc/foolsand_effect.mdl4"] = Vector(0.5, 0.5, 0.5),
+	["models/gstands/stand_acc/foolsand_effect.mdl5"] = Vector(0.8, 0.8, 1),
+	--Collor
+	["models/death/death/death.mdl"] = Vector(0.254901961,0,0.290196078),
```
These lines need the skin value appended (0) or they would not work with the function. Also, where are they used?

ZeroTwo-gStands replied:
These colors are used for the avatars in the Q menu.

copperwarrior commented on lua/autorun/gstands.lua:
```diff
@@ -1,1802 +1,2066 @@
+	--Collor
+	["models/death/death/death.mdl"] = Vector(0.254901961,0,0.290196078),
+	--Collor ser
+	["models/death/death/sercollor.mdl"] = Vector(0.05,0.05,0.05),
```
What is this and where is it used?

ZeroTwo-gStands replied:
These colors are used for the avatars in the Q menu.

copperwarrior commented on lua/autorun/resources.lua:
```diff
@@ -1,858 +1,952 @@
+		"weapons/justice/taunt04.wav",
+	}
+} )
+sound.Add( {
+	name = "world.deploys",
+	channel = CHAN_WEAPON,
+	volume = 1.0,
+	level = 80,
+	sound = {
+		"weapons/world/deploy_01-1.wav",
+		"weapons/world/deploy-02.wav",
+	}
+} )
```
Deploy_01-1 cuts off sharply. Please adjust or remove.

copperwarrior commented on lua/entities/cfhurricane/shared.lua:
```diff
@@ -1,150 +1,150 @@
+			if not(Ent:IsPlayer() or Ent:IsNPC() or Ent:GetClass() == "prop_ragdoll") and Ent != self.Owner then 
+				dmgInfo = DamageInfo()
+				dmgInfo:SetAttacker(damager)
+				dmgInfo:SetDamage(GetConVar("gstands_cfhurricane_damage"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/entities/d13wall/cl_init.lua:
```diff
@@ -12,18 +13,20 @@ function ENT:Initialize()
 	render.FogEnd(5000)
 	render.FogMode(MATERIAL_FOG_LINEAR)
-	render.FogMaxDensity(0.3)
+	for k, ent in pairs(ents.GetAll()) do
+        if ent:GetClass() == "d13scythe"then
+            render.FogMaxDensity(5)
+        else
+            render.FogMaxDensity(0.3)
+        end
+    end
```
Fog is not rendered per-entity, this code block will not do anything except lag.

copperwarrior commented on lua/entities/d13wall/init.lua:
```diff
@@ -3,44 +3,40 @@ AddCSLuaFile("cl_init.lua")
-self.BalloonList = {}
-	for i=0, 10 do
+	self.BalloonList = {}
+	for i=0, 12 do
```
Why the increase?

copperwarrior commented on lua/entities/dbm_waterblast/shared.lua:
```diff
@@ -1,161 +1,161 @@
+					local Ent = tr.Entity
+					dmgInfo = DamageInfo()
+					dmgInfo:SetAttacker(self.Owner or self)
+					dmgInfo:SetDamage(GetConVar("gstands_dark_blue_moon_waterblast"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/entities/death/shared.lua:
```diff
@@ -1,241 +1,298 @@
+ENT.Type = "anim"
+ENT.Base = "base_anim"
+
+ENT.PrintName= "Death"
```
Please revert this to use the proper translation key.

copperwarrior commented on lua/entities/death/shared.lua:
```diff
@@ -1,241 +1,298 @@
+			AddOriginToPVS( self:WorldSpaceCenter() )
+		end
+	end)
+	self:SetSkin(self.Owner:GetInfoNum("gstands_standskin_"..string.Replace(string.Replace("Death 13", " ", ""), "'",""), math.random(0, self:SkinCount() - 1)))
```
Since the localization update, skin convars are set via weapon classname. Please revert this change.

copperwarrior commented on lua/entities/death/shared.lua:
```diff
@@ -1,241 +1,298 @@
+		
+		self:DrawModel() -- Draws Model Client Side
+
+		--if self:GetModel() == "models/d13/d13.mdl" then
+			--halo.Render(haloInfo)
+		--end
```
Please fully remove this code block, instead of just commenting.

copperwarrior commented on lua/entities/death/shared.lua:
```diff
@@ -1,241 +1,298 @@
+				self.StandAura:SetControlPoint(1, gStands.GetStandColor(self:GetModel(), self:GetSkin()))
+		self.BaseClass.Draw(self) -- Overrides Draw 
+		local selfT = { Ents = self }
+		local haloInfo = 
+		{
+			Ents = selfT,
+			Color = gStands.GetStandColorTable(self:GetModel(), self:GetSkin()),
+			Hidden = when_hidden,
+			BlurX = 2,
+			BlurY = 2,
+			DrawPasses = 3,
+			Additive = false,
+			IgnoreZ = true
```
This is only used in the halo render, which is currently unused. Please remove.

copperwarrior commented on lua/entities/emerald_splash_proj/shared.lua:
```diff
@@ -1,156 +1,156 @@
+			local Ent = self.Owner.SplashTrace.Entity
+			dmgInfo = DamageInfo()
+			dmgInfo:SetAttacker(self.Owner or self)
+			dmgInfo:SetDamage(GetConVar("gstands_emerald_splash_damage"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/entities/fool_stand/shared.lua:
```diff
@@ -1,661 +1,663 @@
+		dmginfo:SetAttacker( attacker )
+		
+		dmginfo:SetInflictor( self.Owner:GetActiveWeapon() or self )
+		dmginfo:SetDamage(GetConVar("gstands_the_fool_slash_punch"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/entities/geb/shared.lua:
```diff
@@ -1,555 +1,555 @@
+        dmginfo:SetInflictor( self.Wep )
+
+        if self.Wep:GetState() then
+		dmginfo:SetDamage(GetConVar("gstands_geb_slash_damage"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/entities/geb/shared.lua:
```diff
@@ -1,555 +1,555 @@
+    end
+    end
+    if not self.Owner:gStandsKeyDown("modifierkey2") and !self.Wep:GetState() then
+        self:SetPos(LerpVector(0.5, self.Owner:GetPos() - Vector(0,0,0), self:GetPos()))
```
The 4 unit difference is important due to the shape of the Geb model. It should appear to blend into the floor, not be a blob that sits atop it.

copperwarrior commented on lua/entities/gstands_tscontroller/shared.lua:
```diff
@@ -1,1029 +1,1027 @@
-				table.insert(GetGlobalEntity("Time Stop").StoppedUsers, ply)
-				if CanPlayerTimeStop(ply) then
-					wep.TSReleaseDelay = CurTime() + 1
-					MsgC(Color(255,0,0), ply:GetName().." is entering stopped time with "..v.ClassName.."!\n")
```
This is important debug information for server operators. Please revert this removal.

copperwarrior commented on lua/entities/hermit_whip/shared.lua:
```diff
@@ -1,137 +1,137 @@
+				mult = 3
+			end
+			dmginfo:SetInflictor( self.wep )
+			dmginfo:SetDamage((GetConVar("gstands_hermit_purple_slash"):GetInt()) * mult )
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/entities/sandwave/shared.lua:
```diff
@@ -1,168 +1,168 @@
+			dmginfo:SetDamageType(DMG_CRUSH)
+			
+			dmginfo:SetInflictor( self )
+			dmginfo:SetDamage(GetConVar("gstands_the_fool_sand_wave"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/entities/slc_gib/shared.lua:
```diff
@@ -1,55 +1,57 @@
+		self.LastBlend = self.LastBlend - 0.01
+		self:DrawModel()
+	end 
+	--if IsValid(IsPlayerStandUser(LocalPlayer())) then
```
Please remove unnecessary comments.

copperwarrior commented on lua/entities/sun_laser/shared.lua:
```diff
@@ -1,207 +1,207 @@
+					local Ent = tr.Entity
+					dmgInfo = DamageInfo()
+					dmgInfo:SetAttacker(self.Owner or self)
+					dmgInfo:SetDamage(GetConVar("gstands_the_sun_laser_first"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/entities/thrown_knife/shared.lua:
```diff
@@ -1,167 +1,167 @@
+			self.MarkToDisable = true
+			self.DisableData = data
+			
+			Ent:TakeDamage((GetConVar("gstands_the_world_knife_damage"):GetInt()) , damager, self.Entity)
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/entities/yellowtempgoo/shared.lua:
```diff
@@ -1,145 +1,145 @@
+			self:Remove()
+		end
+		self.DelayDmg = self.DelayDmg or CurTime()
+		if targ:Health() and SERVER and self.DelayDmg <= CurTime() and 	IsValid(self.wep) then
```
Please cleanup whitespace between "and   IsValid("

copperwarrior commented on lua/weapons/gstands_anubis/shared.lua:
```diff
@@ -1,482 +1,533 @@
+		pos = {width - 1550 * mult, height - 235 * mult},
+		color = nocompletegstands,
+	}, 2 * mult, 250)
+end
```
Please use translation keys for this text.

copperwarrior commented on lua/weapons/gstands_anubis/shared.lua:
```diff
@@ -1,482 +1,533 @@
+	timer.Simple(0.1, function() 
+		if self:GetOwner() != nil then
+			if self:GetOwner():IsValid() and SERVER then
+				self:GetOwner():SetHealth(GetConVar("gstands_anubis_heal"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_anubis/shared.lua:
```diff
@@ -1,482 +1,533 @@
+	self.TauntTimer = self.TauntTimer or CurTime()
+	if self.Owner:gStandsKeyDown("taunt") and CurTime() >= self.TauntTimer then
+		if SERVER then
+			self.Owner:EmitSound("weapons/anubis/taunt"..math.random(1, 4)..".wav", 75, 100 )
```
Please use a soundscript instead of random numbers. See: lua/autorun/resources.lua

copperwarrior commented on lua/weapons/gstands_atum/shared.lua:
```diff
@@ -1,717 +1,761 @@
+			font = "gStandsFont",
+			pos = {width - 1550 * mult, height - 235 * mult},
+			color = nocompletegstands,
+		}, 2 * mult, 250)
```
Please use translation keys for this text.

copperwarrior commented on lua/weapons/gstands_atum/shared.lua:
```diff
@@ -1,717 +1,761 @@
+	timer.Simple(0.1, function() 
+		if self:GetOwner() != nil then
+			if self:GetOwner():IsValid() and SERVER then
+				self:GetOwner():SetHealth(GetConVar("gstands_atum_heal"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_atum/shared.lua:
```diff
@@ -1,717 +1,761 @@
+
+local material = Material( "vgui/hud/gstands_hud/crosshair" )
+function SWEP:DoDrawCrosshair(x,y)
+	if IsValid(self.Stand) and IsValid(self.Owner) and IsValid(LocalPlayer()) then
```
Checking validation of LocalPlayer is unnecessary, there should be no case where the crosshair is drawing and the localplayer is invalid.

copperwarrior commented on lua/weapons/gstands_atum/shared.lua:
```diff
@@ -1,717 +1,761 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_atum_heal"):GetInt())
+			end
+		end
+	end)
```
Weapon Initialize is not run when the weapon has an owner. Please use Deploy for setting health.

copperwarrior commented on lua/weapons/gstands_bastet/shared.lua:
```diff
@@ -1,322 +1,426 @@
+SWEP.StandModel 			= "models/props_lab/tpplugholder_single.mdl"
+SWEP.StandModelP 			= "models/props_lab/tpplugholder_single.mdl"
+if CLIENT then
+	SWEP.StandModel = "models/empress/empress.mdl"
```
This is not empress.

copperwarrior commented on lua/weapons/gstands_bastet/shared.lua:
```diff
@@ -1,322 +1,426 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_bastet_heal"):GetInt())
+			end
+		end
+	end)
```
See previous review for Atum.

copperwarrior commented on lua/weapons/gstands_bastet/shared.lua:
```diff
@@ -1,322 +1,426 @@
+			font = "gStandsFont",
+			pos = {width - 145 * mult, height - 190 * mult},
+			color = tcolor,
+		}, 2 * mult, 250)
```
What is "standard" and "rozetka"?
Also, translation keys must start with #

copperwarrior commented on lua/weapons/gstands_bastet/shared.lua:
```diff
@@ -1,322 +1,426 @@
+			pos = {width - 1550 * mult, height - 235 * mult},
+			color = nocompletegstands,
+		}, 2 * mult, 250)
+	end
```
See previous review in Atum.

copperwarrior commented on lua/weapons/gstands_cream/shared.lua:
```diff
@@ -1,753 +1,812 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_cream_heal"):GetInt())
+			end
+		end
+	end)
```
See previous review for Atum.

copperwarrior commented on lua/weapons/gstands_cream/shared.lua:
```diff
@@ -1,753 +1,812 @@
+			font = "gStandsFont",
+			pos = {width - 1550 * mult, height - 235 * mult},
+			color = nocompletegstands,
+		}, 2 * mult, 250)
```
See previous review for Atum.

copperwarrior commented on lua/weapons/gstands_cream/shared.lua:
```diff
@@ -1,753 +1,812 @@
+			text = "Особый",
+			font = "gStandsFont",
+			pos = {width - 145 * mult, height - 190 * mult},
+			color = tcolor,
```
Please use translation keys for this text.

copperwarrior commented on lua/weapons/gstands_dark_blue_moon/shared.lua:
```diff
@@ -1,867 +1,877 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_dark_blue_moon_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous review for Atum.

copperwarrior commented on lua/weapons/gstands_dark_blue_moon/shared.lua:
```diff
@@ -1,867 +1,877 @@
+		dmginfo:SetAttacker( attacker )
+		
+		dmginfo:SetInflictor( self.Owner )
+		dmginfo:SetDamage(GetConVar("gstands_dark_blue_moon_slash_damage"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_death/shared.lua:
```diff
@@ -1,669 +1,668 @@
+SWEP.StandModelP = "models/d13/d13.mdl"
+if CLIENT then
+	SWEP.StandModel = "models/hpworld2/hpworld2.mdl"
+end
```
This is not Hermit Purple 2.

copperwarrior commented on lua/weapons/gstands_death/shared.lua:
```diff
@@ -1,669 +1,668 @@
+local Shine = Sound("weapons/d13/shine.wav")
+local Romantic = Sound("weapons/d13/romantic-to-die.wav")
+local Throw = Sound("weapons/d13/throw.wav")
+local Taunt = Sound("weapons/d13/are-you-dumb-.wav")
```
Filename is "are-you-dumb.wav". Please remove trailing dash.

copperwarrior commented on lua/weapons/gstands_death/shared.lua:
```diff
@@ -1,669 +1,668 @@
+	local Dreamland = GetDreamland(self.Owner)
+	self:SetDream(Dreamland)
+	if !Dreamland:IsValid() then
+		self.Owner:ChatPrint("Карта слишком маленькая для Death 13!")
```
Please revert this change to the translation key.

copperwarrior commented on lua/weapons/gstands_death/shared.lua:
```diff
@@ -1,669 +1,668 @@
+			end
+            if hit then
+				for k, ply in pairs( player.GetAll() ) do
+					ply:ChatPrint( ent:Name().." потерял сознание!" )
```
Please revert this change to use the translation key.

copperwarrior commented on lua/weapons/gstands_death/shared.lua:
```diff
@@ -1,669 +1,668 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_death_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous review for Atum.

copperwarrior commented on lua/weapons/gstands_death/shared.lua:
```diff
@@ -1,669 +1,668 @@
+	if SERVER and self.Owner:gStandsKeyDown("modifierkey1") and self:GetStand():GetModel() != "models/d13/balloon.mdl" and CurTime() >= self.Scythethrow then
+        self:SetNextPrimaryFire( CurTime() + 1 )
+        self:GetStand():ResetSequence(self:GetStand():LookupSequence("scythethrow"))
+		Dreamland:SetPlaybackRate(3)
```
Why is the playback rate changed here?

copperwarrior commented on lua/weapons/gstands_death/shared.lua:
```diff
@@ -1,669 +1,668 @@
+		end)
+		--timer.Simple(3,function()
+			--Dreamland:SetPlaybackRate(1)
+		--end)
```
Please remove or replace commented code.

copperwarrior commented on lua/weapons/gstands_death/shared.lua:
```diff
@@ -1,669 +1,668 @@
+
+--Screw this entire section, it only works like 20% of the time. Basically, meant to catch every time the weapon is not in your hands in order to remove the stand.
+function SWEP:OnDrop()
+	self:GetOwner():SetModelScale(1)
```
Please remove artifact of removed code.

copperwarrior commented on lua/weapons/gstands_death/shared.lua:
```diff
@@ -1,669 +1,668 @@
+		Dreamland:Remove()
+	end
+	self:SetInDoDoDo(false)
+	--self.Owner:RemoveFlags(FL_ATCONTROLS)
```
This is important to reinstate player control after death.

copperwarrior commented on lua/weapons/gstands_ebonydevil/shared.lua:
```diff
@@ -1,554 +1,566 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_ebony_devil_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous review for Atum.

copperwarrior commented on lua/weapons/gstands_emperor/shared.lua:
```diff
@@ -1,488 +1,519 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_emperor_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous review for Atum.

copperwarrior commented on lua/weapons/gstands_empress/shared.lua:
```diff
@@ -1,343 +1,354 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_empress_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous review for Atum.

copperwarrior commented on lua/weapons/gstands_fool/shared.lua:
```diff
@@ -1,648 +1,655 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_the_fool_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous review for Atum.

copperwarrior commented on lua/weapons/gstands_fool/shared.lua:
```diff
@@ -1,648 +1,655 @@
+				dmginfo:SetAttacker( attacker )
+				
+				dmginfo:SetInflictor( self.Owner:GetActiveWeapon() or self )
+				dmginfo:SetDamage(GetConVar("gstands_the_fool_dome_punch"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_geb/shared.lua:
```diff
@@ -1,652 +1,660 @@
+SWEP.StandModelP 			= "models/gstands/stands/geb.mdl"
+if CLIENT then
+	SWEP.StandModel = "models/player/dbm/dbm.mdl"
+end
```
This is not Dark Blue Moon.

copperwarrior commented on lua/weapons/gstands_geb/shared.lua:
```diff
@@ -1,652 +1,660 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_geb_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_geb/shared.lua:
```diff
@@ -1,652 +1,660 @@
+end
+
+function SWEP:OnRemove()
+
```
Why remove this?

copperwarrior commented on lua/weapons/gstands_hanged_man/shared.lua:
```diff
@@ -1,679 +1,688 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_hanged_man_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_hanged_man/shared.lua:
```diff
@@ -1,679 +1,688 @@
+		dmginfo:SetAttacker( attacker )
+		dmginfo:SetDamageType(DMG_SLASH)
+		dmginfo:SetInflictor( self )
+		dmginfo:SetDamage(GetConVar("gstands_hanged_man_neck_damage"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_hermit_purple/shared.lua:
```diff
@@ -1,977 +1,989 @@
+SWEP.StandModelP 				= "models/hpworld/hpworld.mdl"
+if CLIENT then
+	SWEP.StandModel = "models/hdm/hdm.mdl"
+end
```
This is not Hanged Man.

copperwarrior commented on lua/weapons/gstands_hermit_purple/shared.lua:
```diff
@@ -1,977 +1,989 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_hermit_purple_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_hermit_purple/shared.lua:
```diff
@@ -1,977 +1,989 @@
+		dmginfo:SetDamageType(DMG_SHOCK)
+		
+		dmginfo:SetInflictor( self )
+		dmginfo:SetDamage(GetConVar("gstands_hermit_purple_donut_punch"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_hermit_purple_2/shared.lua:
```diff
@@ -1,729 +1,741 @@
+SWEP.StandModelP 				= "models/hpworld/hpworld.mdl"
+if CLIENT then
+	SWEP.StandModel = "models/hpworld2/hpworld2.mdl"
+end
```
There is no hpworld2.mdl.

copperwarrior commented on lua/weapons/gstands_hermit_purple_2/shared.lua:
```diff
@@ -1,729 +1,741 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_hermit_purple_two_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_hierophant/shared.lua:
```diff
@@ -1,1044 +1,1050 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_hierophant_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_hierophant/shared.lua:
```diff
@@ -1,1044 +1,1050 @@
+		self.BarrierAvg = Vector(0,0,0)
+	end
+	self:EndAttack()
+
```
Why remove this?

copperwarrior commented on lua/weapons/gstands_highpriestess/shared.lua:
```diff
@@ -1,644 +1,652 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_high_priestess_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_highpriestess/shared.lua:
```diff
@@ -1,644 +1,652 @@
+		
+		dmginfo:SetInflictor( self )
+		if !self.Stand:GetGround() then
+			dmginfo:SetDamage(GetConVar("gstands_highpriestess_bite_damage"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_horus/shared.lua:
```diff
@@ -1,546 +1,552 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_horus_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_judgement/shared.lua:
```diff
@@ -1,532 +1,553 @@
+local robo4 = Sound("weapons/jgm/robo/robo_4.wav")
+local robo5 = Sound("weapons/jgm/robo/robo_5.wav")
+local laugh = Sound("jgm.laugh")
+local block = Sound("weapons/jgm/laugh/block1.wav")
```
This file does not exist.

copperwarrior commented on lua/weapons/gstands_judgement/shared.lua:
```diff
@@ -1,532 +1,553 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_judgement_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_judgement/shared.lua:
```diff
@@ -1,532 +1,553 @@
+			local clr = (self.Color)
+			local h,s,v = ColorToHSV(clr)
+			h = h - 180
+			clr = HSVToColor(h,1,1)
```
This block does nothing in this context.

copperwarrior commented on lua/weapons/gstands_judgement/shared.lua:
```diff
@@ -1,532 +1,553 @@
+				if IsValid(self) and IsValid(self.Stand) then
+					self.Stand:EmitStandSound(robo5)
+				end
+			end)
```
These sound clips have music playing in the background and cut off harshly. Please fix or remove.

copperwarrior commented on lua/weapons/gstands_judgement/shared.lua:
```diff
@@ -1,532 +1,553 @@
+        if ( !IsValid( attacker ) ) then attacker = self.Owner end
+        dmginfo:SetAttacker( attacker )
+        dmginfo:SetInflictor( self )
+		dmginfo:SetDamage(GetConVar("gstands_judgement_donut_damage"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_justice/shared.lua:
```diff
@@ -1,1024 +1,1031 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_justice_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous review for Atum.

copperwarrior commented on lua/weapons/gstands_kfar/shared.lua:
```diff
@@ -1,398 +1,410 @@
+		end
+	end)   
+end
+
```
Please see previous review for Atum.

copperwarrior commented on lua/weapons/gstands_khnum/shared.lua:
```diff
@@ -1,191 +1,245 @@
+                self:GetOwner():SetMaxHealth(GetConVar("gstands_khnum_heal"):GetInt())
+            end
+		end
+	end)
```
Please see previous review for Atum.

copperwarrior commented on lua/weapons/gstands_khnum/shared.lua:
```diff
@@ -1,191 +1,245 @@
+			font = "gStandsFont",
+			pos = {width - 1550 * mult, height - 235 * mult},
+			color = nocompletegstands,
+		}, 2 * mult, 250)
```
Please see previous review for Atum.

copperwarrior commented on lua/weapons/gstands_khnum/shared.lua:
```diff
@@ -1,191 +1,245 @@
+local corner_left  	= Material(base.."corner_left")
+local corner_right  = Material(base.."corner_right")
+function SWEP:DrawHUD()
+	--if IsValid(self.Stand) then
```
Please remove unused comment.

copperwarrior commented on lua/weapons/gstands_khnum/shared.lua:
```diff
@@ -1,191 +1,245 @@
-        end
-        --Set time until power punch.
-        self:SetNextSecondaryFire( CurTime() + (0.1 * self.Speed) )
-    else
```
Why remove this? Please revert.

copperwarrior commented on lua/weapons/gstands_lovers/shared.lua:
```diff
@@ -1,569 +1,576 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_lovers_heal"):GetInt())
+			end
+		end
+ 	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_magicians_red/shared.lua:
```diff
@@ -1,1071 +1,1079 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_magicians_red_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous review for Atum.

copperwarrior commented on lua/weapons/gstands_magicians_red/shared.lua:
```diff
@@ -1,1071 +1,1079 @@
+		local dmginfo = DamageInfo()		
+		dmginfo:SetAttacker( self.Owner )
+		dmginfo:SetInflictor( self )
+		dmginfo:SetDamage(GetConVar("gstands_magicians_red_kick_damage"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_osiris/shared.lua:
```diff
@@ -1,304 +1,354 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_osiris_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_osiris/shared.lua:
```diff
@@ -1,304 +1,354 @@
+			font = "gStandsFont",
+			pos = {width - 1550 * mult, height - 235 * mult},
+			color = nocompletegstands,
+		}, 2 * mult, 250)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_sethan/shared.lua:
```diff
@@ -1,393 +1,471 @@
+SWEP.UseHands = true
+if CLIENT then
+	SWEP.StandModel = "models/hdm/hdm.mdl"
+end
```
This is not Hanged Man.

copperwarrior commented on lua/weapons/gstands_sethan/shared.lua:
```diff
@@ -1,393 +1,471 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_sethan_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_sethan/shared.lua:
```diff
@@ -1,393 +1,471 @@
+			font = "gStandsFont",
+			pos = {width - 1550 * mult, height - 235 * mult},
+			color = nocompletegstands,
+		}, 2 * mult, 250)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_sethan/shared.lua:
```diff
@@ -1,393 +1,471 @@
+			dmginfo:SetAttacker( attacker )
+			
+			dmginfo:SetInflictor( self )
+			dmginfo:SetDamage(GetConVar("gstands_sethan_punch"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_silver_chariot/shared.lua:
```diff
@@ -1,1793 +1,1836 @@
+				end
+			end
+		end
+	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_silver_chariot/shared.lua:
```diff
@@ -1,1793 +1,1836 @@
+		
+		dmginfo:SetInflictor( self )
+		
+		dmginfo:SetDamage(GetConVar("gstands_silver_chariot_barrage_no_flame"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+local Deploy = Sound( "platinum.deploy" )
+local SDeploy = Sound( "weapons/platinum/deploy.wav" )
+local Withdraw = Sound( "weapons/platinum/withdraw.wav" )
+local Withdraugh = Sound( "weapons/platinum/o_sound.wav" )
```
File is called "augh.wav"

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_star_platinum_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+			end)
+			LocalPlayer():ChatPrint( "『Star Platinum!』" ) 
+			timer.Simple(1.15, function()
+				LocalPlayer():ChatPrint( "『The World!』" ) 
```
These chat prints are tacky. The sound already plays globally, there is no reason to have it play in chat.

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+				
+				surface.SetMaterial(cooldown_box)
+				if !IsValid(GetGlobalEntity("Time Stop")) then
+					surface.DrawRect(width - (56 * mult) - 32 * mult, height - (56 * mult) - 340 * mult, 40 * mult, math.Clamp(0, 40 * mult, math.Remap(self:GetTimeStopDelay() - CurTime(), GetConVar( "gstands_star_platinum_next_timestop" ):GetFloat(), 0, 0, 40 * mult)))
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+		if (not Pain or Pain <= CurTime()) then
+			if ( IsValid( owner ) && dmginfo:GetDamage() >= 10 && dmginfo:GetDamage() < 300 ) and owner:IsPlayer() and owner:HasWeapon("gstands_star_platinum") then
+				owner:EmitSound("weapons/platinum/pain/pain-0"..math.random(1, 2)..".wav", 75, 100 )
+				SetPain(CurTime()+3.8)
```
Pain sounds are more for the user than the stand. This should be moved to its own separate addon.

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+
+
+function SWEP:StopTime()
+
```
Revert this change. Very important that time stop is not falsely registered when time stop is disabled.

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+				zawaeffect:SetMagnitude(1)
+
+				util.Effect( "tweffect", zawaeffect, true, true)
+
```
There is no "tweffect".

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+
+			self.TimeStopperController:Spawn()
+
+			if IsValid(GetGlobalEntity( "Time Stop" )) then
```
Check validity of self.TimeStopperController instead of getting the global entity reference again.

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+	if IsValid(self.Owner) then
+
+		--self.Owner:EmitSound(StartTimeVoice)
+
```
Please remove commented code.

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+
+		self:SetNextSecondaryFire( CurTime() + 4 )
+
+		self:SetTimeStopDelay(CurTime() + GetConVar( "gstands_star_platinum_next_timestop" ):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+						end
+
+						bearing:SetPos(pos)
+
```
Why remove this line?

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+
+		dmginfo:SetInflictor( self )
+
+		dmginfo:SetDamage(GetConVar("gstands_star_platinum_barrage_damage"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+		self.KillstreakTimer = CurTime() + 5
+
+	--end
+	
```
"Augh" voiceline is meant specifically as an easter egg for killing with the Donut Punch, in reference to killing N'Doul with a single punch. Please remove.

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+
+			dmginfo:SetInflictor( self )
+
+			dmginfo:SetDamage(GetConVar("gstands_star_platinum_punch_damage"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+		end
+
+		if ( SERVER && IsValid( tr.Entity ) and !tr.Entity.IsGrabbed and !(tr.Entity.Owner and tr.Entity.Owner.IsGrabbed) and ( tr.Entity:IsStand() ) ) then
+
```
Please revert. Distance check is important to make sure you can't pull Hierophant's owner out of a room 10 miles away.

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+				filter = {self.Stand.Owner, self.Stand},
+
+				mask = MASK_SHOT_HULL
+
```
Star Finger is hard enough to land, there is no reason to make it harder with TraceLine.

copperwarrior commented on lua/weapons/gstands_star_platinum/shared.lua:
```diff
@@ -1,1567 +1,2791 @@
+
+				self.Owner:EmitStandSound(Withdraw)
+			end)
+
```
Why have the timer?

copperwarrior commented on lua/weapons/gstands_strength/shared.lua:
```diff
@@ -1,577 +1,589 @@
+SWEP.StandModelP 				= "models/sun/gstands_sun.mdl"
+if CLIENT then
+	SWEP.StandModel = "models/hdm/hdm.mdl"
+end
```
This is neither Hanged Man nor The Sun.

copperwarrior commented on lua/weapons/gstands_strength/shared.lua:
```diff
@@ -1,577 +1,589 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_strength_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_temperance/shared.lua:
```diff
@@ -1,711 +1,724 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_yellow_temperence_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_temperance/shared.lua:
```diff
@@ -1,711 +1,724 @@
+			dmginfo:SetAttacker( attacker )
+			local mult = 1
+			dmginfo:SetInflictor( self )
+			dmginfo:SetDamage(GetConVar("gstands_temperance_slash"):GetInt() * mult)
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_tenoresax/shared.lua:
```diff
@@ -1,351 +1,425 @@
+SWEP.gStands_IsThirdPerson = true
+if CLIENT then
+	SWEP.StandModel = "models/player/heg/heg.mdl"
+end
```
This is not Hierophant Green.

copperwarrior commented on lua/weapons/gstands_tenoresax/shared.lua:
```diff
@@ -1,351 +1,425 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_tenore_sax_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_tenoresax/shared.lua:
```diff
@@ -1,351 +1,425 @@
+			pos = {width - 1550 * mult, height - 235 * mult},
+			color = nocompletegstands,
+		}, 2 * mult, 250)
+	--end
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+SWEP.Primary.ClipSize	   = 50
+SWEP.Primary.DefaultClip	= 50
+SWEP.Primary.Automatic	  = true
+SWEP.Primary.Ammo		  	= "ammo_ball_bearing"
```
The World has no ball bearings, this and DrawAmmo are pointless.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+					font = "gStandsFont",
+					pos = {width - 1550 * mult, height - 235 * mult},
+					color = nocompletegstands,
+				}, 2 * mult, 250)
```
Please see previous code review for Atum.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+			end
+		end
+	end
+end)
```
Pain sounds should be implemented in a separate addon. Base gStands should focus on primarily the stands.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+		net.Receive("world.PlaySound2", 
+			function()
+				LocalPlayer():ChatPrint( "『Tokiwa Mukidas』" )
+				LocalPlayer():EmitSound(StartTime) 
```
These chatprints are tacky. The sound already plays globally, there is no reason to put it in chat. Please remove.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+		self.Owner:SetWalkSpeed(450)
+		self.Owner:SetRunSpeed(450)
+		self.Owner:SetJumpPower(260)
+	end
```
The World itself makes no changes to the user. Please remove.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+				if !v:Alive() then
+					v:Lock()
+					net.Start("world.Blind")
+					net.Send(v)
```
Blindness is handled in the time stop controller, not in the individual stands.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+				self.Owner:EmitSound("weapons/world/road_roller-0"..math.random(1, 2)..".wav", 60, 100 )
+			else 
+				self.roller:SetModel("models/player/dio/roadroller.mdl")
+				self.Owner:EmitSound("weapons/world/road_roller-0"..math.random(1, 2)..".wav", 60, 100 )
```
Please use a soundscript instead of random numbers. See: lua/autorun/resources.lua

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+		self.Owner:SetAnimation( PLAYER_RELOAD )
+		self.Owner:SetAngles(ang)
+		if SERVER then
+			self.Owner:EmitSound("weapons/world/wry-0"..math.random(1, 2)..".wav", 60, 100 )
```
Please use a soundscript instead of random numbers. See: lua/autorun/resources.lua

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+					self:SetNextSecondaryFire( CurTime() )
+				end
+				else
+				self.Owner:ChatPrint("Ударный режим.") 
```
Please use translation keys for this text.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+			if GetConVar( "gstands_the_world_limit" ):GetInt() == 1 then
+				timer.Create( "stopMax"..self.Owner:GetName(), GetConVar( "gstands_the_world_timestop_length" ):GetFloat(), 1,
+				function() 
+				if GetConVar( "gstands_the_world_limit" ):GetInt() == 1 then
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+						util.Effect( "tweffect", zawaeffect, true, true)
+					end
+				end
+			end)
```
There is no tweffect or tweffect2

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+				table.insert(self.TimeStopperController.StoppedUsers, self.Owner)
+			end
+			self.TimeStopperController:Activate()
+			self:SetTimeStopDelay(CurTime() + GetConVar( "gstands_the_world_timestop_length" ):GetFloat())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+			end
+			self.TimeStopperController = GetGlobalEntity( "Time Stop" )
+			self.TimeStopperController:Spawn()
+			if IsValid(GetGlobalEntity( "gstands_tscontroller" )) then
```
`GetGlobalEntity("Time Stop")`

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+		if SERVER then
+			GetConVar( "gstands_time_stop" ):SetInt(0)
+		end
+		self:SetNextPrimaryFire(CurTime())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+			end
+			timer.Create("knifethrowtimer",0.02,10,function() self:KnifeThrow() end)
+			self.BearingTimer = CurTime() + 1
+		end
```
Knives should not need ammo. Dio was pulling hundreds of them out of nowhere.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+						self.Stand:ResetSequence( "standidle" ) 
+					end
+				end)
+				timer.Create( "sutahprachina"..self.Owner:GetName(), 2.2, 1, function() 
```
This is The World. Naming the timer this would cause conflicts.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+				net.Broadcast()
+			end
+			self.TSReleaseDelay = self.TSReleaseDelay or CurTime()
+			elseif SERVER and IsValid(GetGlobalEntity( "Time Stop" )) and ((GetConVar( "gstands_time_stop" ):GetInt() == 1 and CurTime() >= self.TSReleaseDelay) or (!GetConVar( "gstands_star_platinum_limit" ):GetBool() and GetGlobalEntity( "Time Stop" ).Owner and CurTime() >= self.TSReleaseDelay)) then
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.
Also, use the right limit convar.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+						if IsValid(self.Stand) and self:GetOwner():Alive() then
+							self.Stand:ResetSequence( self.Stand:LookupSequence("SWIMMING_FIST") )
+						end
+					end)
```
This is already done when the hold type is set to "stando". Please remove.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+				self.Stand:ResetSequence( self.Stand:LookupSequence("spinkick") )
+				timer.Simple(0.7, function()
+					self.Stand:ResetSequence( self.Stand:LookupSequence("SWIMMING_FIST") )
+				end)
```
See previous review.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+			end
+			local anim = "punch2"
+			if SERVER then
+				self.Owner:EmitSound("weapons/world/shine-0"..math.random(1, 2)..".wav", 60, 99 )
```
Please use a soundscript instead of random numbers. See: lua/autorun/resources.lua

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+							self:SetHoldType( "stando")
+							self.CurrentAnim = "swimming_fist"
+							self.Stand:ResetSequence( self.Stand:LookupSequence("swimming_fist") ) 
+						end
```
Animation resets should be handled by the stand code, by setting hold type to "stando".

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+		dmginfo:SetAttacker( attacker )
+		dmginfo:SetDamageForce(self.Stand:GetAngles():Forward() * 750)
+		dmginfo:SetInflictor( self )
+		dmginfo:SetDamage(GetConVar("gstands_the_world_barrage_damage"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+		local attacker = self.Owner
+		dmginfo:SetAttacker( attacker )
+		dmginfo:SetInflictor( self )
+		dmginfo:SetDamage(GetConVar("gstands_the_world_punch_damage"):GetInt())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+						local phys = bearing:GetPhysicsObject()
+						local vec = self.Owner:GetAimVector()
+						if self.Stand:GetBlockCap() and IsValid(self.Stand:GetAITarget()) then
+							vec = (self.Stand.AITarget:WorldSpaceCenter() - bearing:GetPos()):GetNormalized()
```
Please revert. Knives should not take ammunition, or only be thrown one at a time.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+local Shine = Sound("weapons/world/shine.wav")
+local Bakana = Sound("weapons/world/bakana.wav")
+local zawarudoda = Sound("weapons/world/deploy-02.wav")
+local SDeploy = Sound("weapons/world/sdep.wav")
```
This file does not exist.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+local zawarudoda = Sound("weapons/world/deploy-02.wav")
+local SDeploy = Sound("weapons/world/sdep.wav")
+local Deploy = Sound("weapons/world/deploy_01.wav")
+local Tomanei = Sound("weapons/world/stoptimes.wav")
```
This file does not exist.

copperwarrior commented on lua/weapons/gstands_the_world/shared.lua:
```diff
@@ -1,1073 +1,1192 @@
+local SDeploy = Sound("weapons/world/sdep.wav")
+local Deploy = Sound("weapons/world/deploy_01.wav")
+local Tomanei = Sound("weapons/world/stoptimes.wav")
+local STEffect = Sound("weapons/world/stop_time_effect_.wav")
```
`"weapons/world/stop_time_effect.wav"`

copperwarrior commented on lua/weapons/gstands_thoth/shared.lua:
```diff
@@ -1,571 +1,645 @@
+			pos = {width - 1550 * mult, height - 235 * mult},
+			color = nocompletegstands,
+		}, 2 * mult, 250)
+	--end
```
Please see previous review for Atum.

copperwarrior commented on lua/weapons/gstands_towergray/shared.lua:
```diff
@@ -1,481 +1,495 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_tower_of_gray_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous review for Atum.

copperwarrior commented on lua/weapons/gstands_towergray/shared.lua:
```diff
@@ -1,481 +1,495 @@
+			if ( !IsValid( attacker ) ) then attacker = self.Owner end
+			dmginfo:SetAttacker( attacker )
+			dmginfo:SetInflictor( self )
+			dmginfo:SetDamage(GetConVar("gstands_tower_gray_damage"):GetInt() * self:GetTongues()/4)
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on lua/weapons/gstands_wheel_of_fortune/shared.lua:
```diff
@@ -1,899 +1,905 @@
+				self:GetOwner():SetMaxHealth(GetConVar("gstands_wheel_of_fortune_heal"):GetInt())
+			end
+		end
+	end)
```
Please see previous review for Atum.

copperwarrior commented on lua/weapons/gstands_wheel_of_fortune/shared.lua:
```diff
@@ -1,899 +1,905 @@
+	local attacker = self.Owner
+	dmginfo:SetAttacker( attacker )
+	dmginfo:SetInflictor( self )
+	dmginfo:SetDamage( GetConVar("gstands_wheel_of_fortune_shoot"):GetInt() * self:GetConfidence())
```
Please take GetConVar out of runtime code, place it outside of any function and use the returned variable locally.

copperwarrior commented on materials/cable/cable_metalwinch01.vmt:
```diff
@@ -0,0 +1,11 @@
+	"$MaxLight"			0.8
+
+	"$surfaceprop"	"metal"
+}
```
As of the latest update, High Priestess uses `effects/hip_cable.vmt`. This file is unnecessary.

copperwarrior commented on materials/cable/hose_black1.vmt:
```diff
@@ -0,0 +1,10 @@
+	"$nocull"			1
+	"$MinLight"			0.3
+	"$MaxLight"			0.8
+}
```
Where is this used?

copperwarrior commented on materials/cable/hose_orange1.vmt:
```diff
@@ -0,0 +1,10 @@
+	"$nocull"			1
+	"$MinLight"			0.3
+	"$MaxLight"			0.7
+}
```
Where is this used?

copperwarrior reviewed (changes requested):
Please address these issues with the pull request. When they are addressed, notify me and I will check through again.
Additional issues:

- The World and Star Platinum have double spaced lines added. Please remove these, as they increase file size for no functional gain.
 - Please go through every added material and make sure that what is added is used in gStands. There is no reason to include files that are not used.
 - For all of the damage and health convars, I think it is better if instead the user could adjust the individual stand stats with a menu. For now, please remove the damage and health convars, I will make a system for changing stats and implement it into both main and the alpha.

ZeroTwo-gStands committed:
Update shared.lua

ZeroTwo-gStands committed:
Update shared.lua

ZeroTwo-gStands committed:
Fix conc. error and upgrade 

Minor improvements to unfinished Stands (Cosmetic) and console bugs fix. I also fixed the avatars.

ZeroTwo-gStands committed:
Fix render and Cons. error

render and cons. error fix

ZeroTwo-gStands committed:
unnecessary files

unnecessary files

ZeroTwo-gStands committed:
unnecessary files

unnecessary files

ZeroTwo-gStands committed:
unnecessary files

unnecessary files

ZeroTwo-gStands committed:
New sounds for Star Platinum and The World

New sounds for Star Platinum and The World

ZeroTwo-gStands committed:
Update shared.lua
